### PR TITLE
Model hub tests

### PR DESCRIPTION
**Pull request by hendrik-obj**
* add tests for model hub
* several small changes to how ObjectivFrame is instantiated
* moved tests just for bach open taxonomy from bach project.

### CODE REVIEW

File: analysis/bach_open_taxonomy/bach_open_taxonomy/series/series_objectiv.py [hunk 1]
```diff
@@ -639,11 +644,12 @@ def mh(self):
         return ModelHub(self)
 
     @classmethod
-    def from_table(cls,
-                   engine=None,
-                   start_date=None,
-                   end_date=None,
-                   time_aggregation: str = None) -> 'ObjectivFrame':
+    def from_objectiv_data(cls,
```
Comment (hendrik-obj) [inline analysis/bach_open_taxonomy/bach_open_taxonomy/series/series_objectiv.py:647]: note the name change. is that ok?

File: analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def sql(self):
             -- generates a unique number for each session, but not in the right order.
             count(is_start_of_session) over (order by user_id, moment, event_id) as is_one_session
         from session_starts_{{id}}
+        order by user_id, moment, event_id
```
Comment (thijs-obj) [inline analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py:39]: this order by doesn't really do anything. The order in the next select might be different again. Perhaps we should do the ordering there if it's important?

File: analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+"""
+Copyright 2021 Objectiv B.V.
+
+Utilities and a dataset for testing Bach Open Taxonomy.
+
+This file does not contain any test, but having the file's name start with `test_` makes pytest treat it
+as a test file. This makes pytest rewrite the asserts to give clearer errors.
```
Comment (thijs-obj) [inline analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py:7]: Comments doesn't make much sense, there are 0 asserts in this file

File: analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+'''
+
+def get_bt_with_json_data_real() -> DataFrame:
+    bt = get_bt('test_json_table_real', TEST_DATA_JSON_REAL, JSON_COLUMNS_REAL, True)
+    bt['global_contexts'] = bt.global_contexts.astype('jsonb')
+    bt['location_stack'] = bt.location_stack.astype('jsonb')
+    return bt
+
+
+def get_objectiv_frame(time_aggregation=None):
```
Comment (thijs-obj) [inline analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py:62]: this function is never used, is that correct?

**Review by thijs-obj — APPROVED** [review]
Few small comments, nothing too big

File: analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py (hunk 2 above)
Comment (hendrik-obj) [reply analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py]: adjusted this.

File: analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py (hunk 3 above)
Comment (hendrik-obj) [reply analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py]: was copied from bach.. removed comment and changed filename.

File: analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py (hunk 4 above)
Comment (hendrik-obj) [reply analysis/bach_open_taxonomy/tests_bach_open_taxonomy/functional/objectiv_bach/test_data_and_utils.py]: you are absolutely right. forgot to add one quite crucial file to this pr. can you still review `test_model_hub.py`?

File: analysis/bach_open_taxonomy/bach_open_taxonomy/series/series_objectiv.py (hunk 1 above)
Comment (jansentom) [reply analysis/bach_open_taxonomy/bach_open_taxonomy/series/series_objectiv.py:646]: I like it

**Review by jansentom — APPROVED** [review]
Looks nice to me!

File: analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py (hunk 2 above)
Comment (thijs-obj) [reply analysis/bach_open_taxonomy/bach_open_taxonomy/stack/sessionized_data.py]: :+1:

File: bach/tests/functional/bach/test_data_and_utils.py [hunk 5]
```diff
@@ -16,7 +16,7 @@
 from bach import DataFrame, Series
 from bach.types import get_series_type_from_db_dtype
 
-DB_TEST_URL = os.environ.get('OBJ_DB_TEST_URL', 'postgresql://objectiv:@localhost:5432/objectiv')
+DB_TEST_URL = os.environ.get('OBJ_DB_TEST_URL', 'postgresql://:@localhost:5432/postgres')
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_data_and_utils.py:19]: question @hendrik-obj: is this change needed?

I just run the Postgres docker container that is launched by `docker-compose -f docker-compose-dev.yaml up objectiv_postgres`. And now the tests bork because it tries to connect as user `thijs` (that's my local user) instead of `objectiv`.

What does your setup look like?
